Cannot manage robot accounts (forbidden: 201)

**Describe the bug**
We can create robot accounts but the terraform state does not store the information and the apply errors out:
```
╷
│ Error: [ERROR] forbidden: status=201 Created, code=201 
│ If you are using a robot account, this is likely due to RBAC limitations. See: https://github.com/goharbor/community/blob/main/proposals/new/Robot-Account-Expand.md
│ 
│   with harbor_robot_account.robot,
│   on main.tf line 30, in resource "harbor_robot_account" "robot":
│   30: resource "harbor_robot_account" "robot" {
│ 
╵
```

**Steps to reproduce the problem:**
Create a project and a robot for it:
```tf
resource "harbor_project" "project" {
  name = "example"
  public = false
}

resource "harbor_robot_account" "robot" {
  name        = "example"
  description = "example robot"
  level       = "project"
  permissions {
    access {
      action   = "pull"
      resource = "repository"
    }
    access {
      action   = "push"
      resource = "repository"
    }
    kind      = "project"
    namespace = harbor_project.project.name
  }
}
```

**Versions:**
Please specify the versions of following systems.

- harbor version: Version v2.13.2-436918f8
- docker engine version: not involved
- docker-compose version: not involved

**Additional context:**

- **Harbor config files:** I have no access to these
- **Log files:** I also have no access to these

Importing seems to kind of work, but deletion also fails – but does not delete the robots.
However, I haven't tested this systematically.

edit: Also, this is not a robot account, but we are using a local account to authenticate.

**To Reproduce**
See above

**Expected behavior**
The return code 201 indicates success (and indeed, the robots are properly created!), but the terraform provider fails and the robots are not stored inside the state.


**Additional context**
Add any other context about the problem here.

- Provider Version: `3.11.2` (constraint was `~> 3.11.0`)
- Terraform Version: none
- Tofu Version: OpenTofu v1.10.6
- Harbor Version: v2.13.2-436918f8

--- a/client/client.go
+++ b/client/client.go
@@ -117,6 +117,9 @@ func (c *Client) SendRequest(method string, path string, payload interface{}, st
 				}
 			}
 		}
+	}
+
+	if resp.StatusCode == http.StatusForbidden {
 		return "", "", resp.StatusCode, fmt.Errorf("[ERROR] forbidden: status=%s, code=%d \nIf you are using a robot account, this is likely due to RBAC limitations. See: https://github.com/goharbor/community/blob/main/proposals/new/Robot-Account-Expand.md", resp.Status, resp.StatusCode)
 	}
 

